src/HDF5.jl

```diff
@@ -1378,26 +1376,41 @@ function getindex(parent::Union{HDF5File, HDF5Group, HDF5Dataset}, r::HDF5Refere
     h5object(obj_id, parent)
 end
 
+# Helper for reading compound types
+function read_row(io::IO, membertype, membersize)
+    row = Vector{Any}()
```

Comment:
@musm I think you are referring to the deprecation `Array(T, size)` in favor of `Array{T}(size)` in julia 0.6. `Vector{Any}()` should be fine, though `Any[]` is more concise.